experiments/recfields/sigma_vs_crossover.m

```diff
@@ -1,140 +1,131 @@
+      std_std  = zeros(length(sigmas), length(freqs));
+      f        = zeros(length(sigmas),1);
+      wts_mean = zeros(length(sigmas), size(wm,1), size(wm,2));
+
+      p=pt(1:0);
+    end;
+
+    avg_mean(si, :) = am;
+    std_mean(si,:) = sm;
+    std_std(si, :) = ss;
+    f(si) = ft;
+    wts_mean(si,:,:) = wm;
+    p(end+1)=pt;
   end;
   
-    lbls = cell(size(p));
-  for pi=1:length(p)
-      lbls{pi} = sprintf('d_{center}=%.1f%% (%.1fpx); d_{nn} = %.1f%% (%.1fpx)', ...
-                         100*p(pi).avg_dist/p(1).sz(1),      p(pi).avg_dist, ...
-                         100*mean(p(pi).neighbor_dist/p(1).sz(1)), mean(p(pi).neighbor_dist));
-  end;
-
-  %
-  scaling = max(std_mean(:)); % Rescale over all sigmas, such that the scale of response isn't a factor
+  
+  %% Analyze raw data for crossover
   numSigmas = size(std_mean, 1);
-  crossover_cpi = zeros(numSigmas * (numSigmas-1), 1);
-  sigma_pairs = zeros(numSigmas * (numSigmas-1), 2);
+  crossover_cpi = nan(numSigmas * (numSigmas-1), 1);
+  sigma_pairs = nan(numSigmas * (numSigmas-1), 2);
   counter = 1;
   for ii=1:numSigmas
-  	for ij=1:numSigmas
-    if ij == ii
-        continue;
-    end
-  	ratios = std_mean(ii,:)./ std_mean(ij,:); % Check when ratio goes over 1
-        if (ratios(1) > 1)
-            i = 2;
-            while ratios(i) >= 1
-                i = i+1;
-                if i > length(ratios)
-                    i = 1;
-                    break;
-                end
-                
-            end
-        else
-            i = 2;
-            while ratios(i) <= 1
-                i = i+1;
-                if i > length(ratios)
-                    i = 1;
-                    break;
-                end
-            end
-        end
-        if i ~= 1 %this means there was a crossover 
-            crossover_cpi(counter) = cpi(i);
-        end
-        sigma_pairs(counter, :) = [sigmas(ii), sigmas(ij)];
-        counter = counter + 1;
-
+  	for ij=(ii + 1):numSigmas
+      ratios = std_mean(ii,:) ./ std_mean(ij,:); % Check when ratio goes over 1
+      if (ratios(1) > 1)  % detect from high to low crossing
+        compareFn = @(idx) ratios(idx) >= 1;
+      else  % detect low to high crossing
+        compareFn = @(idx) ratios(idx) <= 1;
+      end;
+      
+      % Search for crossover point
+      si = 2;
+      while compareFn(si)
+        si = si+1;
+        if si > length(ratios)  % failure
+          si = 1;
+          break;
+        end                
+      end
+      
+      if si ~= 1 %this means there was a crossover
+        crossover_cpi(counter) = cpi(si);
+      end
+      sigma_pairs(counter, :) = [sigmas(ii), sigmas(ij)];
+      counter = counter + 1;
     end
   end 
   
-  figure; hold on;
-  start = 1;
   
-  for ii=1:numSigmas
-      sp = sigma_pairs(start:start+numSigmas-2, 2);
-      cc = crossover_cpi(start:start+numSigmas-2);
-      
-      sp( all(~cc,2), : ) = []; %get rid of the entries that didn't have crossover
```

Comment:
Yup; you can set that to `nan` and get the same effect.

I also wanted to plot all lines together, so that MATLAB would color-code them. Maybe my version is very old, but all lines come out blue for me ...